File: src/eqn/ast/EqnAstNodeSin.java
package eqn.ast;

import eqn.parser.exception.EqnException;
import org.jetbrains.annotations.NotNull;

public class EqnAstNodeSin extends EqnAstNodePreDefinedUnaryFunction {

    public EqnAstNodeSin(@NotNull EqnAstNode equationNode) throws EqnException {
        super("sin", equationNode);
    }

    @Override
    public double evaluate() throws EqnException {
        return Math.sin(operand().evaluate());
    }
}

File: src/eqn/ast/EqnAstNodeSin.kt
package eqn.ast

import eqn.ast.base.EqnAstNode
import eqn.ast.base.EqnAstNodeUnary
import eqn.parser.exception.EqnException
import kotlin.math.sin

class EqnAstNodeSin(operand: EqnAstNode) : EqnAstNodeUnary("sin", Type.PreDefinedUnaryFunction, PrecedenceType.Function, operand) {
    @Throws(EqnException::class)
    override fun evaluate(arguments: Map<String, Double>?): Double {
        return sin(operand.evaluate(arguments))
    }
}